src/wp-admin/includes/image.php

```diff
@@ -1155,3 +1222,78 @@ function _copy_image_file( $attachment_id ) {
+ * For example an `image/jpeg` should be converted into an `image/jpeg` and `image/webp`. The first type
+ * is considered the primary output type for this image.
+ *
+ * @since 6.0.0
+ *
+ * @param $attachment_id int The attachment ID.
+ * @return array<string, array<string>> An array of valid mime types, where the key is the source file mime type and the
+ *                                      value is one or more mime file types to generate.
+ */
+function wp_upload_image_mime_transforms( $attachment_id ) {
+	$image_mime_transforms = array(
+		'image/jpeg' => array( 'image/jpeg', 'image/webp' ),
+		'image/webp' => array( 'image/webp', 'image/jpeg' ),
+	);
+
+	/**
+	 * Filter to the output mime types for a given input mime type.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $image_mime_transforms A map with the valid mime transforms where the key is the source file mime type
+	 *                                     and the value is one or more mime file types to generate.
+	 * @param int   $attachment_id         The ID of the attachment where the hook was dispatched.
+	 */
+	return (array) apply_filters( 'wp_upload_image_mime_transforms', $image_mime_transforms, $attachment_id );
+}
+
+/**
+ * Extract the primary and additional mime output types for an image from the $image_mime_transforms.
+ *
+ * @since 6.0.0
+ * @access private
+ *
+ * @param string $file          Full path to the image file.
+ * @param int    $attachment_id Attachment ID to process.
+ * @return array<string, array<string>> An array with the primary mime type and the additional mime types.
+ */
+function _wp_get_primary_and_additional_mime_types( $file, $attachment_id ) {
+	$image_mime_transforms = wp_upload_image_mime_transforms( $attachment_id );
+	$original_mime_type    = wp_get_image_mime( $file );
+	$output_mime_types     = isset( $image_mime_transforms[ $original_mime_type ] ) ? $image_mime_transforms[ $original_mime_type ] : array( $original_mime_type );
+
+	// Exclude any output mime types that the system doesn't support.
+	$output_mime_types = array_filter(
+		$output_mime_types,
+		function( $mime_type ) {
+			return wp_image_editor_supports(
+				array(
+					'mime_type' => $mime_type,
+				)
+			);
+		}
+	);
+
+	// Use original mime type as primary mime type, or alternatively the first one.
+	$primary_mime_type_key = array_search( $original_mime_type, $output_mime_types, true );
+	if ( false === $primary_mime_type_key ) {
+		$primary_mime_type_key = 0;
+	}
+	// Split output mime types into primary mime type and additional mime types.
+	$additional_mime_types     = $output_mime_types;
+	list( $primary_mime_type ) = array_splice( $additional_mime_types, $primary_mime_type_key, 1 );
+
+	// Ensure $primary_mime_type is set.
+	if ( empty( $primary_mime_type ) ) {
+		$primary_mime_type = $original_mime_type;
+	}
+	return array_merge(
+		array( $primary_mime_type ),
+		$additional_mime_types
+	);
```

Comment:
Why not return the two values like this?
```suggestion
	return array(
		$primary_mime_type,
		$additional_mime_types,
	);
```

This way any code calling the function can simply use `list( $primary_mime_type, $additional_mime_types ) = ...` and doesn't have to manually extract the primary MIME type from the greater array.

Reply:
implemented in current patch